File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.java
class Solution {
    public int findUnsortedSubarray(int[] nums) {
        
        
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        
        int start = -1;
        for(int i=0; i<nums.length; i++){
            if(nums[i] < max){
                start = i;
            }
            else{
                max = nums[i];
            }
        }
         
        
        int end = 0;
        for(int i=nums.length-1; i>=0; i--){
            if(nums[i] > min){
                end = i;
            }
            else{
                min = nums[i];
            }
        }
        
        return start-end+1;
    }
}

File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
       int start=0, end=-1;
        int max=INT_MIN;
        int min=INT_MAX;
        
        
        for(int i=0; i<nums.size(); i++){
            if(nums[i]<max){
                end=i;
            }
            else{
                max = nums[i];
            }
        }
        
        
        for(int i=nums.size()-1; i>=0; i--){
            if(nums[i]>min){
                start=i;
            }
            else{
                min = nums[i];    
            }
        }
        
        cout<<start<<" "<<end;
        return end-start+1;
    }
};

